site/helm-charts.md

```diff
@@ -0,0 +1,513 @@
+
+Defaults to false.
+
+If set to true, the user must provide the SSL certificate and private key for the operator's external REST interface by specifying the "externalOperatorCert" and "externalOperatorKey" properties.
+
+Example:
+```
+externalRestEnabled: true
+```
+
+#### externalRestHttpsPort
+Specifies node port that should be allocated for the external operator REST https interface.
+
+Only used when externalRestEnabled is true, otherwise ignored.
+
+Defaults to 31001.
+
+Example:
+```
+externalRestHttpsPort: 32009
+```
+
+#### externalOperatorCert
+
+Specifies the user supplied certificate to use for the external operator REST https interface. The value must be a string containing a Base64 encoded PEM certificate. This parameter is required if 'externalRestEnabled' is true, otherwise, it is ignored.
+
+There is no default value.
+
+The helm installation will produce an error, similar to the following, if "externalOperatorCert" is not specified (left blank) and "externalRestEnabled" is true:
+```
+Error: render error in "weblogic-operator/templates/main.yaml": template: weblogic-operator/templates/main.yaml:4:3: executing "weblogic-operator/templates/main.yaml"
+  at <include "operator.va...>: error calling include: template: weblogic-operator/templates/_validate-inputs.tpl:53:4: executing "operator.validateInputs"
+  at <include "operator.re...>: error calling include: template: weblogic-operator/templates/_utils.tpl:137:6: executing "operator.reportValidationErrors"
+  at <fail .validationErro...>: error calling fail: The string property externalOperatorCert must be specified.
+```
+
+Example:
+```
+externalOperatorCert: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUQwakNDQXJxZ0F3S ...
+```
+
+#### externalOperatorKey
+
+Specifies user supplied private key to use for the external operator REST https interface. The value must be a string containing a Base64 encoded PEM key. This parameter is required if 'externalRestEnabled' is true, otherwise, it is ignored.
+
+There is no default value.
+
+The helm installation will produce an error, similar to the following, if "externalOperatorKey" is not specified (left blank) and "externalRestEnabled" is true:
+```
+Error: render error in "weblogic-operator/templates/main.yaml": template: weblogic-operator/templates/main.yaml:4:3: executing "weblogic-operator/templates/main.yaml"
+  at <include "operator.va...>: error calling include: template: weblogic-operator/templates/_validate-inputs.tpl:53:4: executing "operator.validateInputs"
+  at <include "operator.re...>: error calling include: template: weblogic-operator/templates/_utils.tpl:137:6: executing "operator.reportValidationErrors"
+  at <fail .validationErro...>: error calling fail: The string property externalOperatorKey must be specified.
+```
+
+Example:
+```
+externalOperatorKey: QmFnIEF0dHJpYnV0ZXMKICAgIGZyaWVuZGx5TmFtZTogd2VibG9naWMtb3B ...
+```
+
+### Debugging options
```

Comment:
I think we shouldn't document these debugging options - I think they're just for operator developers who need to attach a debugger to the operator.

Ryan, do you agree?

Reply:
Ryan: I'd probably include them if there is a change we would ask customers to try them.

Added Ryan as reviewer